en/guide/routing.md

```diff
@@ -1,14 +1,81 @@
+(messages are not resent if the addressee is not known or is on the original/incoming channel). 
+
+
+## Routing Detail
+
+Systems/components should process a message locally if any of these conditions hold:
+- It is a broadcast message (`target_system` field omitted or zero).
+- The `target_system` matches its system id and `target_component` is broadcast (`target_component` omitted or zero).
+- The `target_system` matches its system id and has the component's `target_component`
+- The `target_system` matches its system id and the component is unknown (i.e. this component has not seen any messages on any link that have the message's `target_system`/`target_component`).
+
+Systems should forward messages to another link if any of these conditions hold:
+- It is a broadcast message (`target_system` field omitted or zero).
+- The `target_system` does not match the system id *and* the system knows the link of the target system (i.e. it has previously seen a message from `target_system` on the link).
+- The `target_system` matches its system id and has a `target_component` field, and the system has seen a message from the `target_system`/`target_component` combination on the link.
+
+> **Note** Non-broadcast messages must only be sent (or forwarded) to known destinations (i.e. a system must previously have received a message from the target system/component). Systems should also clear stored information about a detected system/component if it detects a `SYSTEM_TIME` message with a decrease in `time_boot_ms`, as this indicates that the system has rebooted, and might have different routing information.
+
+Routed messages are *unchanged*; the source and destination ids are the values from the originating system.
+
+## Library Support
+
+### C Library (mavgen)
+
+The generated code for MAVLink v1 has no specific support for routing or working with `target_system` and `target_component`.
+
+The MAVLink v2 generator has been updated to make it easier to get the destination system and component ID from the payload (when these are assigned). 
+Specifically, the `mavlink_msg_entry_t` structure contains the offsets in the payload to the target system and component ids (`target_system_ofs` and `target_system_ofs`, respectively). The MAVLink helper method `const mavlink_msg_entry_t*` [`mavlink_get_msg_entry(uint32_t msgid)`](https://github.com/mavlink/c_library_v2/blob/master/protocol.h) can be used to get this structure from the message id.
+
+
+## MAVLink 2 Routing
+
+Unsigned MAVLink 2 packets are routed in the same way as MAVLink 1 packets.
+Signed packets may have some special routing requirements (see [Routing Signed Packets](#routing_signed_packets) below).
+
+
+## Routing Signed Packets {#routing_signed_packets}
+
+Routing signed packets is largely undefined at time of writing (for example, it is not clear whether a router should simply forward a signed message or decode and recode with its own keys).
```

Comment:
If you decode/encode then it should probably be set with your own system id, which really breaks the routing rules above about not messing with these fields. Not really worth a doc change here, I just wanted to mention it.

Reply:
Yes - thanks. 
I'm thinking along the lines that signed messages should simply be forwarded even if they can't be validated by the routing system. Leaving open for now.